version: '3'
services:
  #################################################  
  gps:
    restart: unless-stopped
    network_mode: host # couldn't pass just port 2947.. 
    environment:
      - TZ=Europe/Oslo
    build:
      context: ./gps
      dockerfile: Dockerfile
    volumes:
      - /home/minus/logs/gps:/gps
  #################################################  
  homeassistant:
    image: homeassistant/raspberrypi3-64-homeassistant:2022.7
    network_mode: host
    restart: unless-stopped
    environment:
      - TZ=Europe/Oslo
    volumes:
      - /home/minus/homeassistant_config:/config
      - /home/minus/logs:/logs
  #################################################
  obd:
    restart: unless-stopped
    devices:
      - '/dev/ttyAMA0:/dev/obd'
    environment:
      - TZ=Europe/Oslo
    build:
      context: ./obd
      dockerfile: Dockerfile
    volumes:
      - /home/minus/logs/obd:/obd
  #################################################
  rclone:
    restart: unless-stopped
    environment:
      - TZ=Europe/Oslo
    build:
      context: ./rclone
      dockerfile: Dockerfile
    volumes:
      - /home/minus/.config/rclone:/config/rclone # Must be configured on host
      - /home/minus/logs:/logs
  #################################################
  presence:
    restart: unless-stopped
    privileged: true # GPIO
    environment:
      - TZ=Europe/Oslo
    build:
      context: ./presence
      dockerfile: Dockerfile
    volumes:
      - /home/minus/logs:/logs
